src/RestControllers/FHIR/FhirMedicationDispenseRestController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+/*
+ * FhirMedicationDispenseRestController.php
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Stephen Nielson <snielson@discoverandchange.com>
+ * @copyright Copyright (c) 2025 Stephen Nielson <snielson@discoverandchange.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\RestControllers\FHIR;
+
+use OpenEMR\FHIR\R4\FHIRResource\FHIRBundle\FHIRBundleEntry;
+use OpenEMR\Services\FHIR\FhirResourcesService;
+use OpenEMR\Services\FHIR\FhirMedicationDispenseService;
+use OpenEMR\RestControllers\RestControllerHelper;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * FHIR MedicationDispense REST Controller
+ */
+class FhirMedicationDispenseRestController
+{
+    /**
+     * @var FhirMedicationDispenseService
+     */
+    private readonly FhirMedicationDispenseService $fhirMedicationDispenseService;
+
+    /**
+     * @var FhirResourcesService
+     */
+    private readonly FhirResourcesService $fhirService;
+
+    public function __construct()
+    {
+        $this->fhirMedicationDispenseService = new FhirMedicationDispenseService();
+        $this->fhirService = new FhirResourcesService();
+    }
+
+    /**
+     * Queries for FHIR fhir MedicationDispenseResource resources using various search parameters.
+     * Search parameters include:
+     * - patient (puuid)
+     * - status (token)
+     * - type (token)
+     * - _id (token)
+     * - _lastUpdated (date)
+     *
+     * @param array <string, int|string> $searchParams - The search parameters
+     * @param string $puuidBind - Optional variable to only allow visibility of the patient with this puuid.
+     * @return Response FHIR bundle with query results, if found
+     */
+    public function getAll(array $searchParams, ?string $puuidBind = null): Response
```

Comment:
don't the newer getAll functions make first param a array with the puuid security stuff in it?

Reply:
looking at code, seems like my above statement not entirely true. just for some of the endpoints.